### DISCUSSION TITLE
How to deploy a graphql nestjs application in vercel?

### DISCUSSION THREAD

**Discussion by siamahnaf** [vercel/community#2112] [Help]
### Summary

Hello, I already find many blog about deploying nestjs into vercel. But my application is graphql nestjs application.

```
{
    "version": 2,
    "builds": [
        {
            "src": "src/main.ts",
            "use": "@vercel/node"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "src/main.ts",
            "methods": [
                "GET",
                "POST",
                "PUT",
                "DELETE"
            ]
        }
    ]
}
```

But I am getting everytime serverless function has been crashed error.

Can anyone help me about deploying nestjs graphql application into vercel.

Note: I try to deploy it from githun repository by importing and deploying project.

**Comment by amyegan** [comment]
Hi @siamahnaf! We need some more information to help you debug this issue. Please share as many of the following as possible:

- the error log details
- a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or your repo if it is public
- an example deployment URL or deployment ID
- screenshots showing the problem, project structure, or any other info you think might help with debugging

**Reply by siamahnaf** [reply]
@amyegan , Thanks for responsing-

Here is my repo- https://github.com/siamahnaf/api-flimybook

This is the url-
https://api-flimybook.vercel.app/

Deployment ID is-
bom1::mcqkj-1681835190979-ee93c8ba8032

**Reply by amyegan** [reply]
Thanks for sharing this. It looks like you're trying to write data, but that isn't working due to the read-only Serverless Function environment. Only [temporary file storage](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage) is available in Serverless Functions so you'll need to set up an external database for long-term data storage. There are integrations available to help make it easier to set up solutions like [Supabase](https://vercel.com/integrations/supabase) or [Fauna](https://vercel.com/integrations/fauna) which have GraphQL options and free plans to help you get started

**Reply by siamahnaf** [reply]
It just a simple app. I am not using any database.

**Reply by jcperez-ch** [reply]
What @amyegan meant is that your simple app is trying to write into vercel file system (perhaps it is generating a schema.gql or something?) basically you can't do that in vercel apps.

**Comment by falkomerr** [comment]
Any updates?
